frontend/lib/src/components/elements/ArrowTable/styled-components.ts

```diff
@@ -36,49 +38,99 @@ export const StyledTableCaption = styled.div(({ theme }) => ({
-  borderBottom: `${theme.sizes.borderWidth} solid ${theme.colors.dataframeBorderColor}`,
+const styleCellFunction = (
+  theme: EmotionTheme,
+  border: Arrow.BorderMode = Arrow.BorderMode.ALL
+): CSSObject => ({
+  // Only have borders on the bottom and right of each cell.
+  borderBottom:
+    border !== Arrow.BorderMode.NONE
+      ? `${theme.sizes.borderWidth} solid ${theme.colors.dataframeBorderColor}`
+      : "none",
   "tbody tr:last-child &": {
-    borderBottom: "none",
+    // For "all" borders, remove bottom border of last row to prevent double border with
+    // table border. For "horizontal" borders, also remove bottom border of last row
+    // since there's no content after it.
+    borderBottom:
+      border === Arrow.BorderMode.ALL || border === Arrow.BorderMode.HORIZONTAL
+        ? "none"
+        : undefined,
   },
-  borderRight: `${theme.sizes.borderWidth} solid ${theme.colors.dataframeBorderColor}`,
+  borderRight:
+    border === Arrow.BorderMode.ALL
+      ? `${theme.sizes.borderWidth} solid ${theme.colors.dataframeBorderColor}`
+      : "none",
   "&:last-child": {
-    borderRight: "none",
+    borderRight: border === Arrow.BorderMode.ALL ? "none" : undefined,
+    // Remove right padding from last cell when no borders, so that the table aligns
+    // with the rest of the page.
+    paddingRight: border === Arrow.BorderMode.NONE ? "0" : theme.spacing.xs,
   },
   verticalAlign: "middle",
   padding: `${theme.spacing.twoXS} ${theme.spacing.xs}`,
+  // Increase the space between columns when there are no vertical borders.
+  "&:not(:first-child)": {
+    paddingLeft:
+      border === Arrow.BorderMode.NONE ||
+      border === Arrow.BorderMode.HORIZONTAL
+        ? theme.spacing.lg
+        : theme.spacing.xs,
+  },
+  // Remove left padding from first column when no borders, so that the table aligns
+  // with the rest of the page.
+  "&:first-child": {
+    paddingLeft: border === Arrow.BorderMode.NONE ? "0" : theme.spacing.xs,
+  },
   fontWeight: theme.fontWeights.normal,
 })
 
-export const StyledTableCell = styled.td(({ theme }) =>
-  styleCellFunction(theme)
+export const StyledTableCell = styled.td<{ borderMode: Arrow.BorderMode }>(
+  ({ theme, borderMode }) => styleCellFunction(theme, borderMode)
 )
-export const StyledTableCellHeader = styled.th(({ theme }) => ({
-  ...styleCellFunction(theme),
+export const StyledTableCellHeader = styled.th<{
+  borderMode: Arrow.BorderMode
+}>(({ theme, borderMode }) => ({
+  ...styleCellFunction(theme, borderMode),
   textAlign: "inherit",
   color: theme.colors.fadedText60,
-  paddingLeft: theme.spacing.sm,
+  // Remove left padding from first cell when no borders, so that the table aligns
+  // with the rest of the page.
+  "&:first-child": {
+    paddingLeft: borderMode === Arrow.BorderMode.NONE ? "0" : theme.spacing.sm,
+  },
+  // Increase the space between columns when there are no vertical borders.
+  "&:not(:first-child)": {
```

Comment:
Can we also use `first-of-type` here? `:first-child` is causing a console error in dev mode:

> The pseudo class ":first-child" is potentially unsafe when doing server-side rendering. Try changing it to ":first-of-type".

We can safely ignore this error since we are not doing server-side rendering, but it's still a bit annoying :(